.github/workflows/filename_workflow.yml

```diff
@@ -0,0 +1,45 @@
+name: aws-api-filename-test
+on:
+  workflow_dispatch: # allow workflow to be manually triggered
+  repository_dispatch:
+    types: [trigger-profiler]
+concurrency:
+  # cancel the current running workflow from the same branch, PR when a new workflow is triggered
+  # when the trigger is not a PR but a push, it will use the commit sha to generate the concurrency group
+  # {{ github.workflow }}: the workflow name is used to generate the concurrency group. This allows you to have more than one workflows
+  # {{ github.ref_type }}: the type of Git ref object created in the repository. Can be either branch or tag
+  # {{ github.event.pull_request.number}}: get PR number
+  # {{ github.sha }}: full commit sha
+  # credit: https://github.com/Sage-Bionetworks-Workflows/sagetasks/blob/main/.github/workflows/ci.yml
+  group: >-
+    ${{ github.workflow }}-${{ github.ref_type }}-
+    ${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+jobs:
+  Benchmarch-Latency:
+    runs-on: ubuntu-22.04-4core-16GBRAM-150GBSSD
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+          cache: 'pip' # caching pip dependencies
+
+      - name: Activate a virtual environment
+        run: |
+          pip install --upgrade pip
+          python3.10 -m venv venv
+          source venv/bin/activate
+
+      - name: Install dependencies
+        run: pip3 install -r requirements.txt
+
+      - name: Run all benchmark tests
+        env:
+          SYNAPSE_AUTH_TOKEN: ${{ secrets.SCHEMATIC_TESTER_SYNAPSE_TOKEN }}
+          SERVICE_ACCOUNT_CREDS: ${{ secrets.SERVICE_ACCOUNT_CREDS }}
+        run:
```

Comment:
@GiaJordan Have you tested run this workflow? If you are doing: `python3 APITests/filename_manifest_validate.py` here, then in the ` _save_manifest` function, probably shouldn't be doing: `self.manifest_path = os.path.join(cwd, f"APITests/test_manifests/filename_{self.num_files}_manifest_{uuid.uuid4()}.csv")`?